Enterprise-CMCS/macpro-onemac
# Two Column Details

Story: https://qmacbis.atlassian.net/browse/OY2-24007

### Details

This PR turns the single-column of detail items on a details page into a two-column view.

Before:
<img width="600" alt="Screen Shot 2023-05-31 at 3 33 03 PM" src="https://github.com/Enterprise-CMCS/macpro-onemac/assets/4022304/2ebe671e-e0bb-4924-9e8e-d103ca6bb83c">

After:
<img width="600" alt="Screen Shot 2023-05-31 at 3 32 30 PM" src="https://github.com/Enterprise-CMCS/macpro-onemac/assets/4022304/37425541-b61d-4b83-88e3-730947d3eed7">

### Changes

- Use of `Grid` from Material-UI to handle the CSS and logic for sizing and scaling
- Use of `Review` remains from the original implementation -- is wrapped in grid structure

### Implementation Notes

- Due to how we configure and display our details sections -- namely, as arrays of items to be displayed rather than an object with named attributes -- it's an expensive effort to process the array and display items in specific portions of the grid as requested in the ticket's AC. This affects the ability to accurately place the divider, and accurately showcase items in the columns defined in the AC.
- The left edge of the right-hand card does not align with the grid due to a difference in Material-UI's breakpoint/scaling system and our own. This can be considered a visual bug and we can make at ticket to address this by [customizing MUI's idea of breakpoints and spacing.](https://v4.mui.com/customization/breakpoints/)

### Test Plan

1. Navigate to a package's details page
2. Ensure the view is as described by the AC (sans the exception in Dev Notes)
3. Ensure this is consistent across various package types' detail pages
